src/test/java/teammates/test/pageobjects/AppPage.java

```diff
@@ -735,20 +735,26 @@ private AppPage verifyHtml(By by, String filePath, boolean isWithRetry) throws I
         boolean isPart = by != null;
         String actual = getPageSource(by);
         try {
+            int maxRetryCount = 5;
+            int waitDuration = 1000;
             String expected = FileHelper.readFile(filePath);
             expected = HtmlHelper.injectTestProperties(expected);
-            if (isWithRetry) {
-                int maxRetryCount = 5;
-                int waitDuration = 1000;
-                for (int i = 0; i < maxRetryCount; i++) {
-                    if (HtmlHelper.areSameHtml(expected, actual, isPart)) {
-                        break;
-                    }
-                    ThreadHelper.waitFor(waitDuration);
-                    actual = getPageSource(by);
+            
+            // The check is done multiple times with waiting times in between to account for
+            // certain elements to finish loading (e.g ajax load, panel collapsing/expanding).
+            for (int i = 0; i < maxRetryCount; i++) {
+                if (i == maxRetryCount - 1) {
+                    // Last retry count: do one last attempt and if it still fails,
+                    // throw assertion error and show the differences
+                    HtmlHelper.assertSameHtml(expected, actual, isPart);
+                    break;
+                }
+                if (HtmlHelper.areSameHtml(expected, actual, isPart)) {
```

Comment:
Why would we need an assert check if it already passes? The flow is like this:
- Try 1, passes -> all is well
- Try 1, fails -> wait for 1 sec, get the actual page source again, go to try 2
- ...
- Final try, passes -> all is well
- Final try, fails -> throw assertion error and display diff as how we know it